File: src/main/java/org/example/eventhub/dto/user/UserFilter.java
package org.example.eventhub.dto.user;

import org.example.eventhub.enums.Role;

import java.time.LocalDateTime;

public record UserFilter(
    String username,
    Role role,
    LocalDateTime fromCreatedAt,
    LocalDateTime toCreatedAt,

    Integer minOrganizedEvents,
    Integer maxOrganizedEvents //TODO сортировку мб?
) {
}

File: src/main/java/org/example/eventhub/dto/user/UserFilter.kt
package org.example.eventhub.dto.user;

import java.time.LocalDateTime;
import org.example.eventhub.enums.Role;

public record UserFilter(
        String username,
        Role role,
        LocalDateTime fromCreatedAt,
        LocalDateTime toCreatedAt,
        Integer minOrganizedEvents,
        Integer maxOrganizedEvents // TODO сортировку мб?
        ) {}
